test/integration-serial/serial_deploy_test.rb

```diff
@@ -525,15 +525,66 @@ def test_batch_dry_run_apply_failure_falls_back_to_individual_resource_dry_run_v
   end
 
   def test_batch_dry_run_apply_success_precludes_individual_resource_dry_run_validation
-    Krane::KubernetesResource.any_instance.expects(:validate_definition).with do |kwargs|
-      kwargs[:kubectl].nil? && !kwargs[:dry_run]
+    Krane::KubernetesResource.any_instance.expects(:validate_definition).with { |params| params[:dry_run] == false }
+    result = deploy_fixtures("hello-cloud", subset: %w(secret.yml))
+    assert_deploy_success(result)
+    assert_logs_match_all([
+      "Result: SUCCESS",
+      "Successfully deployed 1 resource",
+    ], in_order: true)
+  end
+
+  def test_resources_with_side_effect_inducing_webhooks_are_not_batched_server_side_dry_run
+    resp = mutating_webhook_fixture(File.join(fixture_path("mutating_webhook_configurations"), "ingress_hook.json"))
+    Krane::ClusterResourceDiscovery.any_instance.expects(:fetch_mutating_webhook_configurations).returns(resp)
```

Comment:
I didn't realize I could just set `failurePolicy: ignore` with a small timeout to circumvent the need for an actual webhook server. I've changed the tests to actually deploy a real `MutatingWebhookConfiguration` and use that to test things out. 

Worth noting that the partitioning logic is only really necessary for `v1beta1` resources, since `v1` only allows (and requires) either `None` or `NoneOnDryRun` to be specified for the `sideEffects` field.